src/Pomelo.Extensions.Caching.MySql/MySqlQueries.cs

```diff
@@ -1,94 +1,89 @@
+			"FROM INFORMATION_SCHEMA.TABLES " +
+			"WHERE TABLE_SCHEMA = '{0}' " +
+			"AND TABLE_NAME = '{1}'";
+
+		private const string UpdateCacheItemFormat =
+		"UPDATE {0} " +
+		"SET ExpiresAtTime = " +
+			"(CASE " +
+				"WHEN TIME_TO_SEC(TIMEDIFF(@UtcNow, AbsoluteExpiration)) <= SlidingExpirationInSeconds " +
+					"THEN AbsoluteExpiration " +
+				"ELSE " +
+					"DATE_ADD(@UtcNow, INTERVAL SlidingExpirationInSeconds SECOND) " +
+			"END) " +
+		"WHERE Id = @Id " +
+		"AND @UtcNow <= ExpiresAtTime " +
+		"AND SlidingExpirationInSeconds IS NOT NULL " +
+		"AND (AbsoluteExpiration IS NULL OR AbsoluteExpiration <> ExpiresAtTime);";
+
+		private const string GetCacheItemFormat =
+			"SELECT Id, ExpiresAtTime, SlidingExpirationInSeconds, AbsoluteExpiration, Value " +
+			"FROM {0} WHERE Id = @Id AND @UtcNow <= ExpiresAtTime;";
+
+		private const string SetCacheItemFormat =
+			"SET @ExpiresAtTime = " +
+			"(CASE " +
+					"WHEN (@SlidingExpirationInSeconds IS NUll) " +
+						"THEN @AbsoluteExpiration " +
+					"ELSE " +
+						"DATE_ADD(@UtcNow, INTERVAL @SlidingExpirationInSeconds SECOND) " +
+			"END);" +
+			 "INSERT INTO {0} " +
+			 "(Id, Value, ExpiresAtTime, SlidingExpirationInSeconds, AbsoluteExpiration) " +
+			 "VALUES (@Id, @Value, @ExpiresAtTime, @SlidingExpirationInSeconds, @AbsoluteExpiration)" +
+			" ON DUPLICATE KEY " +
+					  "UPDATE Value = @Value, ExpiresAtTime = @ExpiresAtTime," +
+					  "SlidingExpirationInSeconds = @SlidingExpirationInSeconds, AbsoluteExpiration = @AbsoluteExpiration " +
+					  ";";
+
+		private const string DeleteCacheItemFormat = "DELETE FROM {0} WHERE Id = @Id";
+
+		public const string DeleteExpiredCacheItemsFormat = "DELETE FROM {0} WHERE @UtcNow > ExpiresAtTime";
+
+		public MySqlQueries(string schemaName, string tableName)
+		{
+			var tableNameWithSchema = string.Format(
+				"{0}.{1}", DelimitIdentifier(schemaName), DelimitIdentifier(tableName));
+
+			// when retrieving an item, we do an UPDATE first and then a SELECT
+			GetCacheItem = string.Format(UpdateCacheItemFormat + GetCacheItemFormat, tableNameWithSchema);
+			GetCacheItemWithoutValue = string.Format(UpdateCacheItemFormat, tableNameWithSchema);
+			DeleteCacheItem = string.Format(DeleteCacheItemFormat, tableNameWithSchema);
+			DeleteExpiredCacheItems = string.Format(DeleteExpiredCacheItemsFormat, tableNameWithSchema);
+			SetCacheItem = string.Format(SetCacheItemFormat, tableNameWithSchema);
+			TableInfo = string.Format(TableInfoFormat, EscapeLiteral(schemaName), EscapeLiteral(tableName));
+		}
+
+		public string TableInfo { get; }
+
+		public string GetCacheItem { get; }
+
+		public string GetCacheItemWithoutValue { get; }
+
+		public string SetCacheItem { get; }
+
+		public string DeleteCacheItem { get; }
+
+		public string DeleteExpiredCacheItems { get; }
+
+		// From EF's MySqlServerQuerySqlGenerator
+		private string DelimitIdentifier(string identifier)
+		{
+			return identifier;
```

Comment:
removed [schema].[table] because Mysql does not use it. If one wishes it it can be added on configuration phase in options